File: types/Product.ts
import { Category } from './Category';
import { CoffeeBean } from './CoffeeBean';
import { CoffeeBrew } from './CoffeeBrews';
import { Stock } from './Stock';

export type Product = {
  id: string;
  name: string;
  category: Category;
  type: string;
  isNowSales: boolean;
  // Coffee
  coffeeBean?: CoffeeBean;
  coffeeBrews?: CoffeeBrew[];
  // Other
  amount?: number;
  stock?: Stock;
};

export type ProductsResponse = {
  products: Product[];
};

export type ProductBody = {
  id: string;
  name: string;
  categoryId: string;
  type: string;
  isNowSales: boolean;
  // Coffee
  coffeeBeanId?: string;
  coffeeBrews?: Array<CoffeeBrew>;
  // Other
  amount?: number;
  stockId?: string;
};

export type ProductPostRequest = Omit<ProductBody, 'id'>;
export type ProductPatchRequest = ProductBody;

File: infra/bundb/dao/product.go
package dao

type Product struct {
	ID           string           `bun:",pk"`
	Name         string           `bun:",notnull"`
	CategoryID   string           `bun:",notnull"`
	Category     *ProductCategory `bun:"rel:has-one"`
	ProductType  uint             `bun:",notnull"`
	IsNowSales   bool             `bun:",notnull,default:true"`
	CoffeeBeanID string
	CoffeeBean   *CoffeeBean `bun:"rel:has-one"`
	Amount       uint
	StockId      string
	Stock        *Stock `bun:"rel:has-one"`
}
